client/src/Components/sitterProfile/bookSitter/BookSitter.js

```diff
@@ -17,89 +27,177 @@ const useStyles = makeStyles({
     color: "white",
     width: "140px",
     height: "45px",
+    margin: "15px",
   },
 });
-function BookSitter() {
-  const classes = useStyles();
-  const date = new Date().getDate() + 1;
-  const month = new Date().getMonth() + 1;
-  const year = new Date().getFullYear();
-  const formattedDate = ("0" + date).slice(-2);
-  const formattedMonth = ("0" + month).slice(-2);
-  const fullDate = year + "-" + formattedMonth + "-" + formattedDate;
+function required(displayName) {
+  return function validateRequired(value) {
+    return value || `${displayName} is required.`;
+  };
+}
+function useOnMount(handler) {
+  return React.useEffect(handler, []);
+}
 
-  const [selectedDate, setSelectedDate] = React.useState(new Date(fullDate));
+function BookSitter({ profile, userProfile }) {
+  const classes = useStyles();
+  const { errors, getValues, handleSubmit, register, setValue } = useForm();
+  const [resErr, setResErr] = useState(false);
+  const [reqSuccess, setReqSuccess] = useState(false);
+  const [resErrMsg, setResErrMsg] = useState("");
   const handleDateChange = (date) => {
-    setSelectedDate(date);
+    setValue("pickupDate", date);
+    values = getValues();
+  };
+  const handleTimeChange = (date) => {
+    setValue("pickupTime", date);
+    values = getValues();
+  };
+  const handleDropoffDateChange = (date) => {
+    setValue("dropoffDate", date);
+    values = getValues();
+  };
+  const handleDropoffTimeChange = (date) => {
```

Comment:
one way to reduce number lines of code i would suggest to create a function 

```
handleChange = (type) => (date) => {
  setValue(type, data); 
  values = getValues();
}

...

<KeyboardDatePicker
  ...
  value={values.dropoffDate}
  onChange={handleChange("dropoffDate")}
  ...
/>
```

bceause the only thing that really is diff between `handleDateChange, handleDropoffDateChange and handleDropoffTimeChange` is the name of the value